### PR TITLE
Saf 156 payment methods changes

### PR DESCRIPTION
**Pull request by Alfr3d**
Screenshot 2021-12-13 at 13 54 29

Now we load payment methods for the terminal via the saferpay API.

### CODE REVIEW

File: src/Factory/HttpClientFactory.php [hunk 1]
```diff
@@ -39,7 +39,7 @@ class HttpClientFactory
      *
      * @return Client
      */
-    public function getClient()
+    public function getClient($headers = [])
```
Comment (margud) [inline src/Factory/HttpClientFactory.php:42]: why add headers here if they are not used?

File: src/Service/SaferPayObtainPaymentMethods.php [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                    'logoUrl' => $paymentMethodObject->LogoUrl,
+                ];
+            }
+        }
+
+        if (!empty($paymentMethodsObject->Wallets)) {
+            foreach ($paymentMethodsObject->Wallets as $wallet) {
+                $paymentMethods[] = [
+                    'paymentMethod' => $wallet->WalletName,
+                    'logoUrl' => $wallet->LogoUrl,
```
Comment (margud) [inline src/Service/SaferPayObtainPaymentMethods.php:67]: i see that you can get logo url here but you never use it? maybe we should start using logos from here?

File: saferpayofficial.php [hunk 3]
```diff
@@ -143,7 +143,10 @@ public function hookPaymentOptions($params)
         if (!$cartService->isCurrencyAvailable($params['cart'])) {
             return;
         }
-
+        /** @var \Invertus\SaferPay\Service\SaferPayObtainPaymentMethods $saferPayObtainPaymentMethods */
```
Comment (margud) [inline saferpayofficial.php:146]: did you added this logic for ps1.6? or we dont support ps1.6?

File: src/Service/SaferPayObtainPaymentMethods.php (hunk 2 above)
Comment (Alfr3d) [reply src/Service/SaferPayObtainPaymentMethods.php:67]: I use it in saferpayofficial.php  line 164

**Pull request merged**